nixos/modules/services/web-apps/hedgedoc.nix

```diff
@@ -1,1001 +1,254 @@
 { config, lib, pkgs, ... }:
-
-with lib;
-
 let
+  inherit (lib) versionAtLeast mkEnableOption mkOption types mdDoc mkIf
+    optionals mkPackageOption mkRemovedOptionModule mkDefault
+    mkRenamedOptionModule literalExpression maintainers;
+
   cfg = config.services.hedgedoc;
 
   # 21.03 will not be an official release - it was instead 21.05.  This
   # versionAtLeast statement remains set to 21.03 for backwards compatibility.
   # See https://github.com/NixOS/nixpkgs/pull/108899 and
   # https://github.com/NixOS/rfcs/blob/master/rfcs/0080-nixos-release-schedule.md.
-  name = if versionAtLeast config.system.stateVersion "21.03"
+  name =
+    if versionAtLeast config.system.stateVersion "21.03"
     then "hedgedoc"
     else "codimd";
 
-  settingsFormat = pkgs.formats.json {};
-
-  prettyJSON = conf:
-    pkgs.runCommandLocal "hedgedoc-config.json" {
-      nativeBuildInputs = [ pkgs.jq ];
-    } ''
-      jq '{production:del(.[]|nulls)|del(.[][]?|nulls)}' \
-        < ${settingsFormat.generate "hedgedoc-ugly.json" cfg.settings} \
-        > $out
-    '';
+  settingsFormat = pkgs.formats.json { };
 in
 {
+  meta.maintainers = with maintainers; [ h7x4 ];
+
   imports = [
     (mkRenamedOptionModule [ "services" "codimd" ] [ "services" "hedgedoc" ])
     (mkRenamedOptionModule
       [ "services" "hedgedoc" "configuration" ] [ "services" "hedgedoc" "settings" ])
+    (mkRenamedOptionModule
+      [ "services" "hedgedoc" "groups" ] [ "users" "users" name "extraGroups" ])
+    (mkRemovedOptionModule [ "services" "hedgedoc" "workDir" ] ''
+      This option has been removed in favor of systemd managing the state directory.
+
+      If you have set this option without specifying `services.settings.uploadsDir`,
+      please move these files to `/var/lib/${name}/uploads`, or set the option to point
+      at the correct location.
+    '')
   ];
 
   options.services.hedgedoc = {
-    enable = mkEnableOption (lib.mdDoc "the HedgeDoc Markdown Editor");
+    enable = mkEnableOption (mdDoc "the HedgeDoc Markdown Editor");
+    package = mkPackageOption pkgs "hedgedoc" { };
```

Comment:
```suggestion
    package = mkPackageOptionMD pkgs "hedgedoc" { };
```

Reply:
Isn't this just equal to `mkPackageOption` nowadays? Based on its comment, it feels like it's supposed to be the other way around, preferring `mkPackageOption` over `mkPackageOptionMD`.

https://github.com/NixOS/nixpkgs/blob/58b1c722f70fff9e967f140f714f19022cc219d1/lib/options.nix#L185-L188

Reply:
This is only equal since a short time and slated for removal but for easier backports it should still be used

Reply:
Sure!